### ISSUE TITLE
glViewportIndexed not supported in OpenGL versions older than 4.1

### ISSUE THREAD

**Issue by marc-cr1810** [mellinoe/veldrid#455]
When trying to run my application on a computer with OpenGL version 3.1. I found that it always crashes when it tries to set the viewport. Looking through the source I found that anytime you try to set the viewport it calls the `glViewportIndexedf` function.
![image](https://user-images.githubusercontent.com/38870664/174976052-c27993e2-6b61-41d7-ab31-6306f93d9cc1.png)
According to OpenGL's own documentation this function is not supported on OpenGL versions below 4.1.
![image](https://user-images.githubusercontent.com/38870664/174976279-5941d6af-be88-4a3a-8c25-7c736bb3cbe9.png)
There should be a check in `Veldrid.OpenGL.OpenGLCommandExecutor.SetViewport` function to detect whether the version is correct to use `glViewportIndexed` or just use `glViewport` which is supported from OpenGL 2.0.
![image](https://user-images.githubusercontent.com/38870664/174977547-2aa3f254-791f-4cc8-9b17-0f4b4062f6fb.png)
![image](https://user-images.githubusercontent.com/38870664/174977374-0c7d1e9d-77e8-46c3-a1a7-c6f3cb9807ae.png)

References:
https://www.khronos.org/registry/OpenGL-Refpages/gl4/html/glViewportIndexed.xhtml
https://www.khronos.org/registry/OpenGL-Refpages/gl4/html/glViewport.xhtml

**Comment by mellinoe** [comment]
Yes, this is an issue. There is already a feature flag (`GraphicsDeviceFeatures.MultipleViewports`) which communicates whether multiple viewports are supported by the device, and this should be reflected correctly already on the GL backend:

```C#
ARB_ViewportArray = IsExtensionSupported("GL_ARB_viewport_array") || GLVersion(4, 1);
```

OpenGLCommandExecutor should consult that feature flag (or check the extension directly) before it calls that function. Additionally, there should probably be a check inside the base `CommandList.SetViewport` method which asserts that you don't try to set a viewport where `index != 0` if the feature isn't supported.
